### PR TITLE
Memory-efficient ANI-1 dataset with MemMap

### PR DESCRIPTION
**Pull request by raimis**
Memory-efficient ANI-1 dataset

### CODE REVIEW

**Comment by raimis** [comment]
**Benchmark**

Load batches of 1024 samples with 4 workers:
```python
from tqdm import tqdm
from torchmdnet.datasets import ANI1
from torch_geometric.loader import DataLoader

ds = ANI1('.')
dl = DataLoader(dataset=ds, batch_size=1024, num_workers=4)

for b in tqdm(dl):
    pass
```

The old implementation:
  - Speed: 25 batch/s
  - Memory: ~16 GB per worker
  
This implementation:
  - Speed: 45 batch/s
  - Memory: ~1 GB per worker

**Comment by raimis** [comment]
@PhilippThoelke can you review?

File: environment.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ dependencies:
   - raimis::pytorch-geometric==2.0.2
   - pytorch-lightning==1.3.8
   - setuptools==59.5.0 # Pin due to https://github.com/pytorch/pytorch/pull/69904
+  - torchmetrics <0.8
```
Comment (PhilippThoelke) [inline environment.yml:22]: Why are you adding torchmetrics here?

Comment (raimis) [reply environment.yml:22]: It is a dependency of PyTorch Lightning, but the latest versions (0.8) is no longer compatible with the old PyL, which we are using.

**Comment by giadefa** [comment]
We should be careful not to save the mem maps on a network drive (e.g.
inside a cluster). Maybe we should allow the user to specify the path?

**Comment by raimis** [comment]
```python
ds = ANI1('.')
```
One can specify a path where the maps are stored.

**Comment by giadefa** [comment]
ok, maybe have something that say that it should be a fast drive.

**Review by PhilippThoelke — APPROVED** [review]

**Pull request merged**